### DISCUSSION TITLE
Add more pagination data to metadata

### DISCUSSION THREAD

**Discussion by ivandoric** [directus/directus#6609] [Feature Requests]
Hey guys,

so I have been having problems when implementing infinite scroll with React Query using REST API. I managed to solve it, kind of, but there a still some bugs. Mainly the problem is that I don't have a single source of truth when trying to figure out on what page I'm currently on. So I had to calculate it on the client side, and this still does not work as expected when changing contexts (like filtering etc.) 

Since a feature like infinite scroll depends on pagination heavily, it would be really nice if you could add some more information to the `metadata` property like `current_page`, `first_page`, `last_page`, `total_pages` etc.

I know  you can calculate all of this on the client side just by using `total_count` and `filter_count` but I think this would be a small but welcomed addition to Directus core. And it would make features like infinite scroll or pagination easier to implement.

Honestly I was a bit baffled to find out that there is not more `metadata` present since most of the CMSs and frameworks like WordPress, Laravel, October CMS, Strapi etc. provide this out of the box.

I'm really new to Directus so please excuse me if I'm missing something and this feature already exists.

**Comment by krazyjakee** [comment]
possible dupe: https://github.com/directus/directus/discussions/4247

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
